---
title: C28198 | Microsoft Docs
ms.custom: ''
ms.date: 11/15/2016
ms.prod: visual-studio-dev14
ms.reviewer: ''
ms.suite: ''
ms.technology:
- vs-devops-test
ms.tgt_pltfrm: ''
ms.topic: article
f1_keywords:
- C28198
helpviewer_keywords:
- C28198
ms.assetid: 8bad4acb-712c-43f5-81d1-45d92092d4c5
caps.latest.revision: 5
author: corob-msft
ms.author: gewarren
manager: ghogen
ms.openlocfilehash: 3d574371401a009b2d5b12708ad347925aca6cc2
ms.sourcegitcommit: 9ceaf69568d61023868ced59108ae4dd46f720ab
ms.translationtype: MT
ms.contentlocale: fr-FR
ms.lasthandoff: 10/12/2018
ms.locfileid: "49292017"
---
# <a name="c28198"></a>C28198
[!INCLUDE[vs2017banner](../includes/vs2017banner.md)]

avertissement C28198 : fuite possible de la mémoire en raison d’une exception.  
  
 Cet avertissement indique que la mémoire allouée n'est pas libéré après qu’une exception est levée. L’instruction à la fin du chemin d’accès peut lever une exception. La mémoire a été passée à une fonction qui peut avoir enregistré une copie à libérer plus tard.  
  
 Cet avertissement est très similaire à avertissement [C28197](../code-quality/c28197.md). Les annotations qui sont recommandées pour une utilisation avec avertissement [C28197](../code-quality/c28197.md) peuvent également être utilisés ici.  
  
## <a name="example"></a>Exemple  
 L'exemple de code suivant génère cet avertissement :  
  
```  
char *p1 = new char[10];  
char *p2 = new char[10];  
  
test(p1);   // does not save a copy of p  
  
delete[] p2;  
delete[] p1;  
```  
  
 L'exemple de code suivant évite cet avertissement :  
  
```  
char *p1 = new char[10];  
char *p2 = NULL;  
  
test(p1);   // does not save a copy of p  
try {  
    p2 = new char[10];  
} catch (std::bad_alloc *e) {  
    // just handle the throw  
    ;  
}  
```



